homepage/src/App.js

```diff
@@ -10,11 +10,12 @@ function App() {
   const [isLoading, setIsLoading] = useState(true);
   const {imageURLS, setImageURLS} = useImage(); 
 
-
-  useEffect(() => {
-    fetchContentfulAPI(imageURLS, setImageURLS, setIsLoading);
-    // eslint-disable-next-line
-  }, []);
+    useEffect(() => {
+      if (Object.keys(imageURLS).length === 0) {
```

Comment:
hmm interesting. so is this what you were talking about? I'll look into the mounting and stuff to better understand.

Reply:
Yup, hooks in general are a super interesting topic to look into. Your changes look good to me!